tx/builder_dynamic_fee.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package tx
+
+import (
+	"encoding/binary"
+	"math/big"
+
+	"github.com/vechain/thor/v2/thor"
+)
+
+// DynFeeBuilder to make it easy to build transaction.
+type DynFeeBuilder struct {
```

Comment:
How about one `TxBuilder` with type as a input method?

Reply:
Yes, this looks simpler. Refactored!